cmd/kubeadm/app/cmd/init.go

```diff
@@ -154,9 +154,9 @@ func newCmdInit(out io.Writer, initOptions *initOptions) *cobra.Command {
 	initRunner.AppendPhase(phases.NewEtcdPhase())
 	initRunner.AppendPhase(phases.NewControlPlanePhase())
 	initRunner.AppendPhase(phases.NewKubeletStartPhase())
-	initRunner.AppendPhase(phases.NewWaitControlPlanePhase())
 	initRunner.AppendPhase(phases.NewUploadConfigPhase())
 	initRunner.AppendPhase(phases.NewUploadCertsPhase())
+	initRunner.AppendPhase(phases.NewWaitControlPlanePhase())
```

Comment:
I'm a little confused, why change the sequence of phases?
I'm afraid executing `UploadConfigPhase` before the control plane is ready will fail.

Reply:
kubelet needs `cluster-admins` role binding exists when requesting pods from kubelet. And `initData.Client()` will make sure the role binding exists before returning api client.

I have reverted `init.go` back, and create a new commit to move this logic into `controlplane/components.go`